### PR TITLE
Update "How to Get PRs Merged" section of CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by MrAlias**
- Correct approval requirement be from a maintainer or approver instead of a maintainer or contributor
- Introduce the defined term "qualified approval"
- Unify the term "trivial change"
- Instead of calling out updates to the CHANGELOG and README, generically say all required CI needs to succeed

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -524,3 +552,6 @@ Emeritus:
 
 See the [community membership document in OpenTelemetry community
 repo](https://github.com/open-telemetry/community/blob/main/community-membership.md).
+
+[Approver]: https://github.com/orgs/open-telemetry/teams/go-approvers
+[Maintainer]: https://github.com/orgs/open-telemetry/teams/go-maintainers
```
Comment (dmathieu) [inline CONTRIBUTING.md:557]: I think these pages are only accessible to members of the organization.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -94,30 +94,58 @@ request ID to the entry you added to `CHANGELOG.md`.
...
+    affiliated with different companies. Two qualified approvals from
+    [Approver]s or [Maintainer]s affiliated with the same company counts as a
+    single qualified approval.
+  * PRs introducing changes that have already been discussed and consensus
+    reached only need one qualified approval. The discussion and resolution
+    needs to be linked to the PR.
+  * Trivial changes[^2] only need one qualified approval.
+
+* All feedback has been addressed.
+  * All PR comments and suggestions are resolved.
```
Comment (Aneurysm9) [inline CONTRIBUTING.md:113]: I don't think this is currently enforced through technical means but can be.  Should we ask for this to be enabled?

Comment (MrAlias) [reply CONTRIBUTING.md:113]: Is there a way to override for the exceptions listed below? IIRC the number of required approved wasn't ignorable.

File: CONTRIBUTING.md (hunk 1 above)
Comment (MrAlias) [reply CONTRIBUTING.md:557]: Sure. Is there a better link to use?

**Review by hanyuancheung — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (pellared) [reply CONTRIBUTING.md:557]: `CODEOWNERS`  file.

Approvers: https://github.com/open-telemetry/opentelemetry-go/blob/b62eb2ca88dfccbffb729f2213a2146ee76dc34e/CODEOWNERS#L15

Maintainers: https://github.com/open-telemetry/opentelemetry-go/blob/b62eb2ca88dfccbffb729f2213a2146ee76dc34e/CODEOWNERS#L17

Comment (MrAlias) [reply CONTRIBUTING.md]: I updated to link to the listing of approves and maintainers in the CONTRIBUTING.md doc itself. PTAL

File: CONTRIBUTING.md (hunk 2 above)
Comment (pellared) [reply CONTRIBUTING.md:113]: I guess that @Aneurysm9 is only asking about getting rid of the following line:

>    * All PR comments and suggestions are resolved.

Comment (MrAlias) [reply CONTRIBUTING.md:113]: Oh gotcha, SGTM 👍

Comment (MrAlias) [reply CONTRIBUTING.md:113]: Do we just need an issue to track this?

Comment (Aneurysm9) [reply CONTRIBUTING.md:113]: I don't think we need to get rid of that line, I'm asking whether we should enable GitHub's enforcement of that requirement.  I think the requirement to resolve all comments doesn't need exceptions in its enforcement mechanism as a maintainer ready to merge a PR can always resolve the outstanding comments if they feel that an exception is warranted.  It adds work, but not a requirement for an additional person to interact as requiring 2 approvals and clearing approvals on changes would.

**Review by dmathieu — APPROVED** [review]

**Review by pellared — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (MrAlias) [reply CONTRIBUTING.md:113]: > I'm asking whether we should enable GitHub's enforcement of that requirement.

Sounds good to me :+1:

Comment (MrAlias) [reply CONTRIBUTING.md:113]: https://github.com/open-telemetry/community/issues/1388

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**